### PR TITLE
Issue #4801: fix some violations from the eleventh part of idea violations

### PR DESCRIPTION
**Pull request by Nimfadora**
Issue #4801

PR not for merge

violations included:
```
UnusedReturnValue
unused
TooBroadCatch
UncheckedExceptionClass
UnusedProperty
TodoComment
ThisEscapedInConstructor
StringTokenizer
VarargParameter
SystemExit
UseOfPropertiesAsHashtable
UnnecessaryFinalOnLocalVariableOrParameter
XmlPathReference
```

### CODE REVIEW

**Comment by romani** [comment]
@Nimfadora ,

there is only one inspection that give violation

> File path resolving in XML (Errors) (351)

suppress: "this inspection is too severely demand all paths to be from repo root, no options to adjust"

**Comment by Nimfadora** [comment]
@romani I have dozens of violations on my local if enable all checks from this report

**Comment by romani** [comment]
@Nimfadora , please recheck why TC is not showing them.
If still a problem ...please share report yourself on github.io.

**Comment by Nimfadora** [comment]
@romani on the inspections page uncheck `inspections with new problems only` and you will see all of them

**Comment by romani** [comment]
> Unnecessary 'final' on local variable or parameter

suppress: "it conflicts with our FinalLocalVariable"

> Method can be void

just fix it.

> Unused declaration

Make settings like:
screenshotscope
and suppress the only violation by suppression javadoc tag.

**Comment by romani** [comment]
> Varargs method

suppress: "we like varargs methods, we use modern jdk"

> Call to 'System.exit()' or related methods

the only case - suppress by javadoc tag .

> Use of Properties object as a Hashtable

the only case - suppress by javadoc tag.

> File path resolving in XML 

disable inspection: "too much false positives"

**Comment by romani** [comment]
Tests are failed

**Comment by Nimfadora** [comment]
@rnveach something wrong with travis:
```
* What went wrong:
Could not resolve all files for configuration ':sbe-tool:compileClasspath'.
> Could not find org.agrona:agrona:0.9.7-SNAPSHOT.
  Searched in the following locations:
      file:/home/travis/.m2/repository/org/agrona/agrona/0.9.7-SNAPSHOT/maven-metadata.xml
      file:/home/travis/.m2/repository/org/agrona/agrona/0.9.7-SNAPSHOT/agrona-0.9.7-SNAPSHOT.pom
      file:/home/travis/.m2/repository/org/agrona/agrona/0.9.7-SNAPSHOT/agrona-0.9.7-SNAPSHOT.jar
      https://repo1.maven.org/maven2/org/agrona/agrona/0.9.7-SNAPSHOT/maven-metadata.xml
      https://repo1.maven.org/maven2/org/agrona/agrona/0.9.7-SNAPSHOT/agrona-0.9.7-SNAPSHOT.pom
      https://repo1.maven.org/maven2/org/agrona/agrona/0.9.7-SNAPSHOT/agrona-0.9.7-SNAPSHOT.jar
  Required by:
      project :sbe-tool
```

File: config/intellij-idea-inspections.xml [hunk 1]
```diff
@@ -2134,6 +2136,16 @@
...
+                <!-- we use it in Main class -->
+                <option value="CallToSystemExit" />
+                <!-- we use it for testing purposes -->
+                <option value="unused" />
+                <!-- we use it for testing purposes -->
+                <option value="UseOfPropertiesAsHashtable" />
+                <!-- till #4861, #4862, #4863, #4864, #4866 -->
+                <option value="ThisEscapedInObjectConstruction" />
+                <!-- generates false-positive in CyclomaticComplexityCheck -->
+                <option value="MismatchedQueryAndUpdateOfCollection" />
```
Comment (romani) [inline config/intellij-idea-inspections.xml]: why you allow `MismatchedQueryAndUpdateOfCollection` to suppress , and do not use this in this PR.
If not required now - it is not required.

File: src/test/java/com/puppycrawl/tools/checkstyle/utils/ModuleReflectionUtilsTest.java [hunk 2]
```diff
@@ -140,6 +140,7 @@ protected InvalidNonAutomaticBeanClass() {
         public abstract void method();
     }
 
+    /** @noinspection unused */
```
Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/utils/ModuleReflectionUtilsTest.java]: please make javadoc for this class, to explain reason of it. Suppressions for unused looks weird.

Comment (Nimfadora) [inline src/test/java/com/puppycrawl/tools/checkstyle/utils/ModuleReflectionUtilsTest.java]: done

File: config/intellij-idea-inspections.xml (hunk 1 above)
Comment (Nimfadora) [inline config/intellij-idea-inspections.xml]: done

**Comment by Nimfadora** [comment]
@romani could you please relaunch TC build, I has timed out

**Comment by rnveach** [comment]
@Nimfadora done.

**Pull request merged**